### PR TITLE
add MPI linking info in install documentation

### PR DESCRIPTION
**Pull request by PrometheusPi**
This pull request adds an additional bullet point for building hdf5 from source. Based on the discussion on #2667, it is a quite common issue to run into linking errors with MPI when configuring HDF5. This can be solved by specifying to use the MPI compilers instead of the default ones (set e.g. by the GCC/ICC module). This new bullet point adds this missing information.

### CODE REVIEW

**Comment by PrometheusPi** [comment]
@sbastrakov You are right, the comma was missing. I fixed it now.

**Comment by ax3l** [comment]
Users usually run those instructions to install a whole bunch of software, on which an export of CC/CXX will influence subsequent unrelated packages as well.

Can you please check if instead the HDF5 configure command knows a CC and CXX argument? Or if simple CC=... CXX=... ./configure serves the same purpose?

**Comment by ax3l** [comment]
@PrometheusPi ping

**Comment by ax3l** [comment]
@PrometheusPi ping

**Comment by ax3l** [comment]
@PrometheusPi ping, please verify and update the PR :)

**Comment by PrometheusPi** [comment]
@ax3l I am on it right now - using the interface directly is a better option. I adjusted the install description accordingly. Right now, I am still testing the entire compile process which takes about an hour. I will inform you if it does not work.

File: INSTALL.rst [hunk 1]
```diff
@@ -234,6 +234,7 @@ HDF5
   - ``make``
   - *optional:* ``make test``
   - ``make install``
+  - If you encounter errors related to linking MPI during ``./configure``, you might try setting the compiler manually via ``./configure --enable-parallel --enable-shared --prefix $HOME/lib/hdf5/  CC=mpicc CXX=mpic++``.
```
Comment (ax3l) [inline INSTALL.rst]: there is an unnecessary double whitespace before CC.

note: when testing, double check `CC` is not `export`-ed from a previous session to the same value :)

Comment (PrometheusPi) [reply INSTALL.rst]: fixed white space

Comment (PrometheusPi) [reply INSTALL.rst]: Install and most tests went well. 
Both are not exported to the previous session (still `gcc` and `g++`).

Comment (ax3l) [reply INSTALL.rst]: thanks for testing!

**Pull request merged**